### PR TITLE
[Jaeger] Add gRPC sampling endpoint

### PR DESCRIPTION
**Pull request by simitt**
## Motivation/summary
Support fetching Jaeger gRPC sampling strategy from APM
server. Only probabilistic sampling is supported. Sampling
rates are configured via Kibana agent config management.

See detailed motivation in https://github.com/elastic/apm-server/issues/3487

## Checklist

~- [ ] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).~
- [x] My code follows the style guidelines of this project (run `make check-full` for static code checks and linting)
- [x] I have rebased my changes on top of the latest master branch

- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing [**unit** tests](https://github.com/elastic/apm-server/blob/master/TESTING.md) pass locally with my changes

- [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)

## How to test these changes
Setup:
* [Download Jaeger agent](https://www.jaegertracing.io/download/) or pull docker image. Start a Jaeger Agent and point to APM Server as Jaeger collector, e.g. `./jaeger-agent --reporter.grpc.host-port=localhost:14250` .
* Start the APM Server,  play with the configurations for `apm-server.jaeger.grpc.sampling` and `apm-server.kibana`. 
* Create agent config entries in Kibana

Test:
Send a sampling request to the running Jaeger agent, that then requests the sampling rate from the APM Server, e.g. `curl localhost:5778/sampling?service=redis`

## Related issues
Implements #3487

### CODE REVIEW

File: beater/jaeger/grpc.go [hunk 1]
```diff
@@ -47,21 +55,92 @@ type grpcCollector struct {
...
+// It fetches the sampling rate from the central configuration management and returns the sampling strategy.
+func (s grpcSampler) GetSamplingStrategy(
+	ctx context.Context,
+	params *api_v2.SamplingStrategyParameters) (*api_v2.SamplingStrategyResponse, error) {
+	gRPCSamplingMonitoringMap.inc(request.IDRequestCount)
+	defer gRPCSamplingMonitoringMap.inc(request.IDResponseCount)
+	samplingRate, err := s.fetchSamplingRate(ctx, params.ServiceName)
+	if err != nil {
+		s.log.With(zap.Error(err)).Error("Fetching sampling rate failed, falling back to default value.")
+		samplingRate = s.defaultRate
```
[the remark this answers is not in the document]
Comment (simitt) [reply beater/jaeger/grpc.go]: That's a good point, I'll rework that.

File: _meta/beat.yml [hunk 2]
```diff
@@ -435,6 +435,16 @@ apm-server:
       # does not apply to Jaeger agents.
       #auth_tag: ""
 
+      # Enable fetching the sampling rate from the gRPC endpoint. Only probabilistic
```
[the remark this answers is not in the document]
Comment (simitt) [reply _meta/beat.yml]: The main difference I see is that for the elastic APM API we offer auth token protection, to ensure no service related information can be leaked. An exception is the RUM endpoint, where we have a special implementation to ensure we only return RUM related information. 
For users upgrading from `7.6` with jaeger enabled, the endpoint would automatically be enabled. 

However, you made me thinking I could apply a similar rule to what we have for RUM, so only configurations for agents starting with `Jaeger` are returned. (We set the agent name to `Jaeger/....` in the otel processor). With this in place I'd be ok to have it enabled by default.

[the remark this answers is not in the document]
Comment (axw) [reply _meta/beat.yml]: Good call. Filtering for `Jaeger` sounds fine to me.

**Comment by simitt** [comment]
@axw once more - thanks for the feedback, I addressed the suggested points.

File: _meta/beat.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (bmorelli25) [reply _meta/beat.yml]: On https://www.elastic.co/subscriptions we list Jaeger intake as open source. Removing the `apm-server.grpc.sampling.default` config value will effectively turn sampling into a basic feature, as users need the APM app to make changes. Not saying we shouldn't do it, but something to be aware of.

[the remark this answers is not in the document]
Comment (graphaelli) [reply _meta/beat.yml]: Centralizing configuration of the sampling rate would be under basic license, the sampling rate could still be defined locally for each jaeger agent.

[the remark this answers is not in the document]
Comment (bmorelli25) [reply _meta/beat.yml]: Ahhh, with `sampler.param`? I didn't consider that. I'll add that to the docs.

[the remark this answers is not in the document]
Comment (simitt) [reply _meta/beat.yml]: Jaeger clients support local settings for sampling or configuring `sampler.type=remote` in which case the sampling rate would be fetched from the APM Server. While fetching agent config remote is the default, local settings in the clients can be configured.

[the remark this answers is not in the document]
Comment (bmorelli25) [reply _meta/beat.yml]: Totally overlooked that. Thanks 😊

File: beater/api/config/agent/handler.go [hunk 3]
```diff
@@ -157,6 +157,7 @@ func buildQuery(c *request.Context) (query agentcfg.Query, err error) {
 		query = agentcfg.NewQuery(
 			params.Get(agentcfg.ServiceName),
 			params.Get(agentcfg.ServiceEnv),
+			nil,
```
[the remark this answers is not in the document]
Comment (simitt) [reply beater/api/config/agent/handler.go]: Don't think it specifically provides value, just tried to not add too many changes to the existing handler. Will change.

**Pull request merged**